src/main/java/me/snoh666/compassTracker/CompassTracker.java

```diff
@@ -24,13 +26,26 @@ public void run() {
 
                 if (closestPlayerUid != null) {
                     Location closestPlayerLoc = getPlayerFromUid(closestPlayerUid).getLocation();
-                    player.setCompassTarget(closestPlayerLoc);
+                       if(player.getInventory().getItemInMainHand().getType().equals(Material.COMPASS)) {
+                           ItemStack compass = player.getInventory().getItemInMainHand();
+                           CompassMeta compassMeta = (CompassMeta) compass.getItemMeta();
+                           if(player.getWorld().getEnvironment() == World.Environment.NETHER) {
+                               compassMeta.setLodestoneTracked(false);
+                               compassMeta.setLodestone(closestPlayerLoc);
+                               compass.setItemMeta(compassMeta);
+                           } else {
+                               compassMeta.setLodestoneTracked(true);
+                               compassMeta.setLodestone(null);
+                               player.setCompassTarget(closestPlayerLoc);
+                           }
+                           compass.setItemMeta(compassMeta);
+                       }
                 }
             }
     }
 
     private UUID getClosestPlayer(Location loc, UUID exceptPlayerId) {
-        if (Objects.requireNonNull(loc.getWorld()).getEnvironment() != World.Environment.NORMAL) {
+        if (Objects.requireNonNull(loc.getWorld()).getEnvironment() != World.Environment.NORMAL && Objects.requireNonNull(loc.getWorld()).getEnvironment() != World.Environment.NETHER) {
```

Comment:
Yeah, me neither, I'm not quite sure if it works there.

Reply:
and are we checking if both players are in the same world environment?

Reply:
Yeah, I totally agree. We should check whether or not both players are in the same world. What do you think we should do in case they are in different worlds? Should we just make it work as a regular compass?

Reply:
No I think let's leave it as it is. So It will leave tracking to last known location of closest palyer in the same world. So It will be logged off location or portal